### ISSUE TITLE
Custom JSON encoding on top level structs

### ISSUE THREAD

**Issue by cblach** [jackc/pgx#1681]
**Describe the bug**
MarshalJSON method is not triggered on the top level struct passed to pgx.

**To Reproduce**

```go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"os"
	"github.com/jackc/pgx/v5"
)

type Parent struct {
 Child Child
}

func (t *Parent) MarshalJSON() ([]byte, error) {
  return []byte(`{"custom":"thing"}`), nil
}

type Child struct{}

func (t Child) MarshalJSON() ([]byte, error) {
  return []byte(`{"someVal": false}`), nil
}

func main() {
	conn, err := pgx.Connect(context.Background(), os.Getenv("DATABASE_URL"))
	if err != nil {
		log.Fatal(err)
	}
	defer conn.Close(context.Background())
	goJSON, err := json.Marshal(&Parent{})
	if err != nil {
		log.Fatalln("json.MarshalJSON", err)
	}
	var pgxJSON []byte
	if err := conn.QueryRow(context.Background(), "select $1::jsonb", &Parent{}).Scan(&pgxJSON); err != nil {
		log.Fatalln("conn.QueryRow", err)
	}
	fmt.Println("=== goJSON ===:\n", string(goJSON))
	fmt.Println("=== pgxJSON ===:\n", string(pgxJSON))
}

```

**Expected behavior**
pgx is expected to produce the same output as go's own JSON encoder and use Parent MarshalJSON method

**Actual behavior**
pgx instead prints the struct members instead of respecting the MarshalJSON method. The Child MarshalJSON method is triggered as expected. Output:

=== goJSON ===:
 {"custom":"thing"}
=== pgxJSON ===:
 {"Child": {"someVal": false}}

**Version**
 - Go: go1.20.4 linux/amd64
 - PostgreSQL: postgres (PostgreSQL) 15.1
 - pgx: github.com/jackc/pgx/v5 v5.4.1

**Comment by jackc** [comment]
Should be fixed as of 524f6611363418799624f2321b158ba405359692.

**Comment by cblach** [comment]
I can confirm you fixed the problem. Thank you!
